browser/ui/brave_browser.cc

```diff
@@ -82,6 +87,26 @@ void BraveBrowser::ScheduleUIUpdate(content::WebContents* source,
 #endif
 }
 
+void BraveBrowser::OnTabClosing(content::WebContents* contents) {
+  Browser::OnTabClosing(contents);
+
+  if (!AreAllTabsSharedPinnedTabs()) {
+    return;
+  }
+
+  if (chrome::FindAllTabbedBrowsersWithProfile(profile()).size() > 1) {
+    base::SequencedTaskRunner::GetCurrentDefault()->PostTask(
+        FROM_HERE, base::BindOnce(
+                       [](base::WeakPtr<BraveBrowser> browser) {
+                         if (browser) {
+                           browser->confirmed_to_close_ = true;
+                           browser->window()->Close();
```

Comment:
Using command api would be better ? - `chrome::CloseWindow(browser);`
and can you add some comment why |confirmed_to_close| should have true?

Reply:
> chrome::CloseWindow(browser);

Good call!

> and can you add some comment why |confirmed_to_close| should have true?

Added comment